sample_factory/algo/sampling/non_batched_sampling.py

```diff
@@ -148,7 +148,9 @@ def curr_actions(self) -> np.ndarray | List | Any:
             return self._process_action_space(actions, is_discrete=False)
         elif isinstance(self.env_info.action_space, gym.spaces.Tuple):
             out_actions = []
-            for split, space in zip(np.split(actions, self.env_info.action_splits), self.env_info.action_space):
+            for split, space in zip(
+                np.split(actions, np.cumsum(self.env_info.action_splits)), self.env_info.action_space
+            ):
```

Comment:
I checked and it looks like on my machine all three versions give different results:

```
>>> import numpy as np
>>> import torch
>>> actions = [1,2,3,4,5,6]
>>> actions_t = torch.tensor(actions)
>>> 
>>> splits = [2,4]
>>> 
>>> np.split(actions, splits)
[array([1, 2]), array([3, 4]), array([5, 6])]
>>> np.split(actions, np.cumsum(splits))
[array([1, 2]), array([3, 4, 5, 6]), array([], dtype=int64)]
>>> torch.split(actions_t, splits)
(tensor([1, 2]), tensor([3, 4, 5, 6]))
```

I think it would be nice to avoid that discrepancy.

Reply:
Here's one possible solution:

```
>>> splits_relative = splits
>>> splits_absolute = np.cumsum(splits_relative)[:-1]
>>> 
>>> np.split(actions, splits_absolute)
[array([1, 2]), array([3, 4, 5, 6])]
```

Reply:
It will also work even if splits is only len=1, so should be safe!

Reply:
That empty array is so strange. Good spot, thanks. I have updated the PR.